### PR TITLE
New nav() api providing legacy support for shiny's tabPanel() logic

### PR DESCRIPTION
**Pull request by cpsievert**
(To be merged after https://github.com/rstudio/htmltools/pull/224)

This PR moves essentially all of the server-side logic behind `{shiny}`'s `tabPanel()` logic and exports it via a new API (in future PRs, we'll be expanding on this new API to provide new functionality). Here's a map from the old API to the new one:

### Creating navs

* `nav()` (i.e., `tabPanel()`)
* `nav_menu()` (i.e., `navbarMenu()`)
* `navs_tab()` (i.e., `tabsetPanel(type = "tabs")`)
* `navs_pill()` (i.e., `tabsetPanel(type = "pill")`)
* `navs_hidden()` (i.e., `tabsetPanel(type = "hidden")`)
* `navs_pill_list()` (i.e., `navlistPanel()`)
* `navs_bar()` (i.e., `navbarPage()`)

Note that the `navs_*()` consumer functions (e.g., `navs_tab()`) generate a result that prints a static HTML page 

```r
tabs <- navs_tab(
  nav("a", "a"),
  nav("b", "b")
)
tabs
```

Screen Shot 2021-05-12 at 6 26 57 PM

But you can also put them inside new page constructor functions that default `theme` to `bs_theme()` (the latest recommended version of Bootstrap).

```r
page_fluid(tabs)
```

### Updating navs

* `nav_select()` (i.e., `updateTabsetPanel()`)
* `nav_insert()` (i.e., `insertTab()`)
* `nav_prepend()` (i.e., `prependTab()`)
* `nav_append()` (i.e., `appendTab()`)
* `nav_remove()` (i.e., `removeTab()`)
* `nav_show()` (i.e., `showTab()`)
* `nav_hide()` (i.e., `hideTab()`)

### CODE REVIEW

File: R/page.R [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#' Creat a Bootstrap page
```
Comment (cpsievert) [inline R/page.R:1]:
```suggestion
#' Create a Bootstrap page
```

File: R/navs-legacy.R [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+}
+
+
+
+# -----------------------------------------------------------------------
+# 'Internal' tabset logic that was pulled directly from shiny/R/bootstrap.R
+#  (with minor modifications)
+# -----------------------------------------------------------------------
+
+navbarPage <- function(title,
```
Comment (wch) [inline R/navs-legacy.R:169]: Rename to `navbar_page`, to make it clear that this is not the same as `shiny::navbarPage`.

File: R/print.R [hunk 3]
```diff
@@ -0,0 +1,34 @@
+fragment <- function(x, page = shiny::fluidPage, theme = bs_global_get() %||% bs_theme(), class = NULL) {
```
Comment (wch) [inline R/print.R:1]: Remove `theme` argument, change `shiny::fluidPage ` to `page_fluid`.

File: R/page.R [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#' Create a page with fluid layout
+#'
+#' Alias for [shiny::fixedPage()] with `theme` defaulting to a newer version
+#' of Bootstrap.
+#'
+#' @export
+#' @inheritParams shiny::fixedPage
+page_fixed <- function(..., title = NULL, theme = bs_theme(), lang = NULL) {
+  browsable2(
```
Comment (wch) [inline R/page.R:39]: The `is_interactive()` check should happen at print time, not creation time.

**Pull request merged**